### ISSUE TITLE
False positive: 'Foreach source to iterate over' with array_rand

### ISSUE THREAD

**Issue by iltar** [kalessil/phpinspectionsea#160]
Check: Type compatibility | Foreach source to iterate over

Given the code:

```php
foreach (array_rand(range('a', 'z'), 10) as $random) {
}
```

This will trigger a "false" positive, as `array_rand` will return an array when the second argument is passed along (yay for php inconsistencies)
![image](https://cloud.githubusercontent.com/assets/1754678/23465527/ce7aee34-fe98-11e6-8615-9ef9525807ea.png)

![image](https://cloud.githubusercontent.com/assets/1754678/23465461/9e9e9652-fe98-11e6-80f7-3b3e76142916.png)
